go/billing/django_utils.py

```diff
@@ -20,6 +22,28 @@ def to_json(self, transactions):
                 for t in simplifier.serialize(transactions))
 
 
+def chunked_query(queryset, items_per_chunk=1000):
+    """Iterate over a queryset using server-side cursors (if possible)
+    and return lists of objects in chunks.
+
+    :type queryset:
+        Django query set.
+    :param queryset:
+        Query to iterate over.
+    :param int items_per_chunk:
+        Number of objects to include in each chunk. Default 1000.
+    """
+    with server_side_cursors(itersize=items_per_chunk):
+        items = []
+        for item in queryset.iterator():
+            items.append(item)
+            if len(items) == items_per_chunk:
+                yield items
+                items = []
```

Comment:
I like
[code]
items = [None]
iterator = queryset.iterator()
while items:
   items = itertools.islice(iterator, itersize)
   yield items
[/code]

but islice isn't exactly the most obvious method :/